core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { forkJoin, Observable } from 'rxjs';
+import { mergeMap, take } from 'rxjs/operators';
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+import { DotCurrentUserService } from '@dotcms/app/api/services/dot-current-user/dot-current-user.service';
+import { DotCurrentUser } from '@dotcms/app/shared/models/dot-current-user/dot-current-user';
+import {
+    DotESContentService,
+    ESOrderDirection
+} from '@dotcms/app/api/services/dot-es-content/dot-es-content.service';
+import { ESContent } from '@dotcms/app/shared/models/dot-es-content/dot-es-content.model';
+import { DotHttpErrorManagerService } from '@dotcms/app/api/services/dot-http-error-manager/dot-http-error-manager.service';
+import { HttpErrorResponse } from '@angular/common/http';
+
+export interface DotPagesState {
+    favoritePages: {
+        items: DotCMSContentlet[];
+        showLoadMoreButton: boolean;
+        total: number;
+    };
+    loggedUserId: string;
+}
+
+const FAVORITE_PAGES_ES_QUERY = `+contentType:dotFavoritePage +deleted:false +working:true`;
+
+@Injectable()
+export class DotPageStore extends ComponentStore<DotPagesState> {
+    private favoritePages = [];
+
+    readonly vm$ = this.state$;
+
+    /** A function that updates the Favorite Pages Items in the state of the store.
+     * @param DotPagesState state
+     * @param DotCMSContentlet[] favoritePages
+     * @memberof DotPageStore
+     */
+    readonly setFavoritePages = this.updater<DotCMSContentlet[]>(
+        (state: DotPagesState, favoritePages: DotCMSContentlet[]) => {
+            return {
+                ...state,
+                favoritePages: {
+                    ...state.favoritePages,
+                    items: [...favoritePages]
+                }
+            };
+        }
+    );
+
+    // EFFECTS
+    readonly getFavoritePages = this.effect((itemsPerPage$: Observable<number>) => {
+        return itemsPerPage$.pipe(
+            mergeMap((itemsPerPage: number) =>
```

Comment:
getFavorite is called several times? mergeMap does not cancel the preview request.

Reply:
changed to switchMap